### ISSUE TITLE
Running service under URL prefix

### ISSUE THREAD

**Issue by matuskosut** [DigitalSlideArchive/digital_slide_archive#263]
Hi,

do you have experience running Digital Slide Archive under URL prefix? 
We are running a common web service on the main domain (e.g. https://custom.domain.com/) and then running Digital Slide Archive within the prefix `/services/dsa` (e.g. https://custom.domain.com/services/dsa/ )

I was following the guide for Girder https://girder.readthedocs.io/en/latest/deployment-alternatives.html#reverse-proxy
but still have issues with some http requests not using URL prefix and just using e.g. `/static/...` instead of `/services/dsa/static/...`. So it seems like most requests are working but some are not following the settings:

```
[server]
api_root = "/services/dsa/api/v1"
static_public_path = "/services/dsa/static"
```

![inspect](https://user-images.githubusercontent.com/10776269/232800022-32fc215b-a773-4c7d-9e82-9b0a8dafafc9.png)

**Comment by manthey** [comment]
We do this on one of our demo servers (and test it on CI).  The nginx fragment is 
```
location /histomicstk/ {
  proxy_set_header X-Forwarded-Host $http_host;
  proxy_set_header X-Forwarded-Server $host;
  proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
  proxy_set_header X-Forwarded-Proto $scheme;
  proxy_pass http://172.17.0.2:8080/;
}
```
In the girder.cfg we have
```
[global]
tools.proxy.on = True
[server]
api_root = "/histomicstk/api/v1"
static_public_path = "/histomicstk/static"
```
The only different I see here is we are only one path deep in the reverse proxy.  Did you include the `tools.proxy.on = True`?

**Comment by manthey** [comment]
That gets created as part of the `girder build` process based on the value of `static_public_path` from the config file.  I think that means that you either changed the config file without rebuilding or rebuilt in a different environment than where the config file is located.

**Comment by matuskosut** [comment]
@manthey oh, I see. I was using docker-compose setup, removing and creating new container every time I change the config. Based on the logs it seems like it is doing some build right after the start.

Is it fine to build for production then? At least that seems to solve css paths.
```
docker exec -ti dsa_girder_1 bash -lc "girder build --mode production"
```

I still get some 502 and cancelled stream, but that I think I need to figure out with NGINX proxy.

![502](https://user-images.githubusercontent.com/10776269/233295007-e65a8993-f1e6-4c57-9635-1ea79dd6d3a9.png)

![stream](https://user-images.githubusercontent.com/10776269/233295439-01413b9f-b676-44f8-a010-f6cbd8602580.png)

**Comment by manthey** [comment]
Yes, doing the build that way is fine.  You can also use a provisioning file to ask it to build the files on `docker-compose up` (e.g., modify provision.yaml or mount your own version of it and have `rebuild-client: True` in that file.
